google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/ConnectionOptions.java

```diff
@@ -370,10 +370,23 @@ private ConnectionOptions(Builder builder) {
                 + matcher.group(Builder.HOST_GROUP);
     this.instanceId = matcher.group(Builder.INSTANCE_GROUP);
     this.databaseName = matcher.group(Builder.DATABASE_GROUP);
-    this.credentials =
-        builder.credentials == null
-            ? getCredentialsService().createCredentials(this.credentialsUrl)
-            : builder.credentials;
+    Credentials credentials = null;
+    if (builder.credentials == null) {
+      try {
+        credentials = getCredentialsService().createCredentials(this.credentialsUrl);
+      } catch (Exception e) {
+        // Ignore the error if the user did not set a credentials URL.
+        // The error is then caused by an invalid environment configuration
+        // and should not be thrown here.
```

Comment:
can we add a test that asserts the correct exception is thrown?

Reply:
I've added a couple of test cases to cover this.